core/notification/src/main/java/edu/stanford/spezi/core/notification/setting/NotificationSettingsRepository.kt

```diff
@@ -0,0 +1,112 @@
+package edu.stanford.spezi.core.notification.setting
+
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.ListenerRegistration
+import edu.stanford.spezi.core.coroutines.di.Dispatching
+import edu.stanford.spezi.core.logging.speziLogger
+import edu.stanford.spezi.module.account.manager.UserSessionManager
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+/**
+ * Repository for managing user's notification settings.
+ */
+class NotificationSettingsRepository @Inject constructor(
+    private val firestore: FirebaseFirestore,
+    private val userSessionManager: UserSessionManager,
+    @Dispatching.IO private val ioDispatcher: CoroutineDispatcher,
+) {
+    private val logger by speziLogger()
+
+    suspend fun observeNotificationSettings(): Flow<Result<NotificationSettings>> =
+        callbackFlow {
+            var listenerRegistration: ListenerRegistration? = null
+            withContext(ioDispatcher) {
+                runCatching {
+                    val userId = userSessionManager.getUserUid() ?: error("User not authenticated")
+                    listenerRegistration = firestore.collection(NOTIFICATION_SETTINGS_PATH)
+                        .document(userId)
+                        .addSnapshotListener { snapshot, exception ->
+                            if (exception != null) {
+                                logger.e(exception) { "Error observing notification settings" }
+                                trySend(Result.failure(exception))
+                            } else {
+                                val notificationSettings =
```

Comment:
I would propose an alternative data structure that is a bit more scalable imo:

```kotlin
enum class NotificationType(val key: String) {
    APPOINTMENT_REMINDERS(key = "receivesAppointmentReminders"),
    MEDICATION_UPDATES(key = "receivesMedicationUpdates"),
    QUESTIONNAIRE_REMINDERS(key = "receivesQuestionnaireReminders"),
    RECOMMENDATION_UPDATES(key = "receivesRecommendationUpdates"),
    VITALS_REMINDERS(key = "receivesVitalsReminders"),
    WEIGHT_ALERTS(key = "receivesWeightAlerts"),
}

data class NotificationSettings(
    private val settings: Map<NotificationType, Boolean>
) : Map<NotificationType, Boolean> by settings {

    override fun get(key: NotificationType): Boolean {
        return settings[key] ?: false
    }
    
    fun update(type: NotificationType, value: Boolean): NotificationSettings {
        val newSettings = settings.toMutableMap()
        newSettings[type] = value
        return NotificationSettings(settings = newSettings.toMap())
    }
}
```

This snippet could be replaced to:

```kotlin
val notificationSettings = NotificationSettings(
    settings = NotificationType.entries.associateWith { type ->
        snapshot?.getBoolean(type.key) ?: false
    }
)
trySend(Result.success(notificationSettings))
```

and in save settings:

```kotlin
suspend fun saveNotificationSettings(
    notificationSettings: NotificationSettings,
): Result<Unit> {
    return withContext(ioDispatcher) {
        runCatching {
            val userId = userSessionManager.getUserUid() ?: error("User not authenticated")
            firestore.collection(NOTIFICATION_SETTINGS_PATH)
                .document(userId)
                .update(notificationSettings.mapKeys { it.key.key })
                .await().let { }
        }
    }
}
```

This can replace the need for the constants, for the mapper for switch type enum.

Feel free to extend the `NotificationType` enum with an additional property for the section (Reminders, updates or Trends), and adjust the ui model to group the settings by sections and use a lazy column in compose. With these adjustments, I think adding or deleting a new setting requires only changing the `NotificationType` and the rest will behave the same